Detectors/MUON/MID/Base/include/MIDBase/DPLSerializer.h

```diff
@@ -0,0 +1,111 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file   MIDBase/DPLSerializer.h
+/// \brief  Templated boost serializer/deserializer for MID vectors in DPL
```

Comment:
Nice ideas.

A few comments related to this file (and maybe others):

a) most certainly it should go into the Framework directory

b) `DPLSerializer` sounds very generic and gives the impression that it could use a mechanism different than boost. Is this true? Otherwise, I suggest to call it `BoostDPLSerializer` or similar.

c) Is it possible to static_assert when some data structure did not implement a boost serializer?

d) Can you say something about the performance gain (difference) compared to ROOT serialization?

Reply:
Ciao Sandro,

Here my answers:

a) I wanted to put this DPL-serializer alongside the one Diego already pushed. Of course I can move it to the Framework folder, maybe in the Utils sub-one.

b) I see the point and indeed this helper uses boost serialisation. I will consider your proposal and change the naming following the advice.

c) Yes, sure. I'll do that.

d) I did not perform tests about that. I can manage to provide some measurements in the next days!

Reply:
Ciao Matthias,

Thanks for the comment.
I will work on b and c and update the branch asap (with some squashes as well).
I'll dig a bit on where to relocate this tool.
For the benchmark Vs ROOT serialisation I have to learn something about it, but I'll try to provide a similar implementation using that.

Just an additional update: I have tested the tool with a workflow and everything works as intended.

Cheers, and happy holidays!

Reply:
Hi @matthiasrichter @gabrielefronze wouldn't it be nice if these things would not be tied too closely to DPL? A DPL API can always be cooked and made dependent on this, but there might be use cases outside this high level API...

Reply:
Hi,

The core serialising method is NOT dependent on DPL, but on boost.
Basically anyone can implement a wrapper of the wanted API that allocates the message space and serialises the container automagically using the core call.
In fact, as you can see, the SerialiseContainer method has no DPL calls within.
Maybe the SerialiseContainer implementation can be moved in DataFormats, so that every "framework" can implement the specialised call.
Let's see what are the others saying about that.
Cheers